### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by rapimo**
simplified code by using postgres internal char function for comparision
standarized extension and Makefile
language_forward.h and language_reverse.h now include
complete functions wich are build using build_lang.sh

### CODE REVIEW

File: sql/pg-language.sql [hunk 1]
```diff
@@ -161,9 +147,7 @@ CREATE OPERATOR <> (
 COMMENT ON OPERATOR <>(language, language) IS 'not equal';
 
 CREATE FUNCTION language_cmp(language, language)
-    RETURNS int4
-    AS '$libdir/language.so'
-    LANGUAGE C IMMUTABLE STRICT;
+RETURNS integer LANGUAGE internal IMMUTABLE AS 'btcharcmp';
```
Comment (einhverfr) [inline sql/pg-language.sql]: I really like this convention.  I was thinking about suggesting it for the other extensions but wanted to get other things done first.

Comment (einhverfr) [reply sql/pg-language.sql]: In fact we could take it further and just use existing internal functions for operators if they are already registered?

Comment (einhverfr) [reply sql/pg-language.sql]: (We don't need to have our own functions at all)

File: build_lang.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+cat > $1 <<-EOF
+/*
+ * code automatically generated with make and build_lang.sh
+ */
+
+#ifndef LANG_REV_H
+#define LANG_REV_H
+#include "language.h"
+static inline language_code
+lang_from_str(const char *str)
```
Comment (zeitgeist) [inline build_lang.sh]: very minor: for sake of coherent naming, I'd propose to call this function `language_from_str`.

**Pull request merged**